transport/http/auth.go

```diff
@@ -2,7 +2,6 @@ package http
 
 import (
 	"context"
-	"net/http"
```

Comment:
nice, i wonder if it was just an oversight that we werent using the smithy request object already

Reply:
Well this is the new signer type from a previous PR, but yes I felt that it should use the smithy transport type instead of the underlying one.